middle_end/flambda2/types/join_levels.ml

```diff
@@ -151,13 +151,21 @@ let join_types ~env_at_fork envs_with_levels =
       Name.Map.merge join_types joined_types (TEL.equations t))
 
 let construct_joined_level envs_with_levels ~env_at_fork ~allowed ~joined_types
-    =
+    ~params =
+  let variable_is_in_new_level var =
+    (* Parameters are already in the resulting environment *)
+    Name_occurrences.mem_var allowed var
+    && not
+         (List.exists
+            (fun param -> Variable.equal var (Bound_parameter.var param))
+            params)
+  in
```

Comment:
This could be quite slow with a large number of parameters, wouldn't it be simpler and faster to remove `params` from `allowed`?